packages/lesswrong/server/migrations/20250503T011345.migrate_user_bookmarks_to_collection.ts

```diff
@@ -0,0 +1,59 @@
+import { randomId } from "@/lib/random";
+import { bulkRawInsert } from "../manualMigrations/migrationUtils";
+import chunk from 'lodash/chunk';
+
+type OldBookmarkRow = {
+  userId: string,
+  postId: string,
+  originalIndex: number,
+}
+
+type BookmarksToInsertData = Omit<DbBookmark, "schemaVersion">
+
+export const up = async ({db}: MigrationContext) => {
+
+  const oldBookmarkRows = await db.manyOrNone<OldBookmarkRow>(`
+    SELECT
+      u._id AS "userId",
+      bookmark_element.el ->> 'postId' AS "postId",
+      bookmark_element.idx AS "originalIndex"
+    FROM
+      "Users" u,
+      LATERAL unnest(u."bookmarkedPostsMetadata") WITH ORDINALITY AS bookmark_element(el, idx)
```

Comment:
sure, I guess.  could've also just fetched the users as single rows and unnested the bookmarks in code?

Reply:
I actually originally did that and then figured that was a bit redundant to unroll/roll/unroll. This way also makes the index be explicit and order more safely preserved